eq-author/src/App/QuestionnairesPage/QuestionnairesTable/Row.js

```diff
@@ -83,49 +105,87 @@ class Row extends React.Component {
     }
   }
 
-  componentDidUpdate() {
-    if (this.props.autoFocus) {
+  componentDidUpdate(prevProps) {
+    if (!prevProps.autoFocus && this.props.autoFocus) {
       this.focusLink();
     }
   }
 
+  handleClick = () => {
+    this.props.history.push(
+      buildQuestionnairePath({
+        questionnaireId: this.props.questionnaire.id,
+      })
+    );
+  };
+
+  handleFocus = () => {
+    this.setState({
+      linkHasFocus: true,
+    });
+  };
+
+  handleBlur = () => {
+    this.setState({
+      linkHasFocus: false,
+    });
+  };
+
+  handleButtonFocus = e => {
+    this.setState({
+      linkHasFocus: false,
+    });
+    e.stopPropagation();
+  };
+
   render() {
-    const { questionnaire, ...rest } = this.props;
+    const { questionnaire, onDeleteQuestionnaire, odd, ...rest } = this.props;
+
     return (
       <FadeTransition {...rest} exit>
-        <TR innerRef={this.rowRef}>
-          <TD>
-            <TruncatedQuestionnaireLink
+        <TR
+          innerRef={this.rowRef}
+          odd={odd}
+          onFocus={this.handleFocus}
+          onBlur={this.handleBlur}
+          linkHasFocus={this.state.linkHasFocus}
+        >
+          <TD onClick={this.handleClick}>
+            <QuestionnaireLink
               data-test="anchor-questionnaire-title"
               questionnaire={questionnaire}
-              title={questionnaire.displayName}
+              title={questionnaire.title}
             >
-              {questionnaire.displayName}
-            </TruncatedQuestionnaireLink>
+              {questionnaire.shortTitle && (
+                <ShortTitle>{questionnaire.shortTitle}</ShortTitle>
+              )}
+              <Truncated>{questionnaire.title}</Truncated>
+            </QuestionnaireLink>
           </TD>
-          <TD>
+          <TD onClick={this.handleClick}>
             <FormattedDate date={questionnaire.createdAt} />
           </TD>
+          <TD onClick={this.handleClick}>{questionnaire.createdBy.name}</TD>
           <TD>
-            <Truncated>{questionnaire.createdBy.name || "Unknown"}</Truncated>
-          </TD>
-          <TD textAlign="center">
-            <ButtonGroup>
-              <DuplicateButton
-                data-test="btn-duplicate-questionnaire"
-                onClick={this.handleDuplicateQuestionnaire}
-              />
-              <IconButtonDelete
-                hideText
-                data-test="btn-delete-questionnaire"
-                onClick={this.handleOpenDeleteQuestionnaireDialog}
-              />
-            </ButtonGroup>
+            <div onFocus={this.handleButtonFocus}>
+              <ButtonGroup>
+                <DuplicateButton
+                  data-test="btn-duplicate-questionnaire"
+                  onClick={this.handleDuplicateQuestionnaire}
+                  style={{ marginRight: "0.5em" }}
```

Comment:
Create a styled component for this (eg. `DuplicateQuestionnaireButton`) and add the margin to that.

Reply:
Done.